src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -470,6 +475,27 @@ public void testBuildName() throws Exception {
   }
 
   @Test
+  public void testBuildWithPull() throws Exception {
+    final String dockerDirectory = Resources.getResource("dockerDirectory").getPath();
+    final String pullMsg = "Pulling from";
+
+    // Build once to make sure we have cached images.
+    sut.build(Paths.get(dockerDirectory));
+
+    // Build again with PULL set, and verify we pulled the base image
+    final AtomicBoolean pulled = new AtomicBoolean(false);
+    sut.build(Paths.get(dockerDirectory), "test", new ProgressHandler() {
+      @Override
+      public void progress(ProgressMessage message) throws DockerException {
+        if (message.stream().contains(pullMsg)) {
+          pulled.set(true);
+        }
+      }
+    }, PULL);
+    assertTrue(pulled.get());
```

Comment:
wow thanks for looking into this.

On Tue, Aug 4, 2015 at 11:00 AM, Ryan Culbertson <notifications@github.com>
wrote:

> In src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
> <https://github.com/spotify/docker-client/pull/217#discussion_r36198157>:
>
> > +    final String pullMsg = "Pulling from";
> > +
> > +    // Build once to make sure we have cached images.
> > +    sut.build(Paths.get(dockerDirectory));
> > +
> > +    // Build again with PULL set, and verify we pulled the base image
> > +    final AtomicBoolean pulled = new AtomicBoolean(false);
> > +    sut.build(Paths.get(dockerDirectory), "test", new ProgressHandler() {
> > +      @Override
> > +      public void progress(ProgressMessage message) throws DockerException {
> > +        if (message.stream().contains(pullMsg)) {
> > +          pulled.set(true);
> > +        }
> > +      }
> > +    }, PULL);
> > +    assertTrue(pulled.get());
>
> I verified with curl that 1.6.2 isn't sending those extra fields. Just
> wanted to make sure it wasn't our deserializer dropping fields (which I
> think we've seen before).
>
> Build output from 1.7.1:
> https://gist.github.com/rculbertson/0794926b2dfb2bbe1272
>
> Build output from 1.6.2:
> https://gist.github.com/rculbertson/f304b6d43637fff6185c
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/docker-client/pull/217/files#r36198157>.
>



-- 
David Xia
Software Engineer
dxia@spotify.com